Automattic/wp-calypso
# Domains: Improve domain connection instructions

As detailed in pcYYhz-xH-p2, some users have problems after connecting domains to WPCOM: when users have domains with connected services (especially email) and change their name servers to ours - as we recommend doing - these services will get disrupted if they hadn't copied DNS records to our side beforehand.

This PR follows p2MSmN-92G-p2's suggestion and updates the instructions to warn users that, in order to not disrupt their connected services, they should copy their DNS records to our infrastructure before updating name servers. This PR also updates various stylings and layouts in the connect a domain setup.

### Changes proposed in this Pull Request

The main update of this PR is the addition of a foldable card with these DNS explanations in the first step of the suggested setup. The other styling and layout differences are:
- There are breadcrumbs now
- Background is now white instead of light gray
- Layout is two columns instead of one
- Transfer nudge card's background is light gray instead of white
- The copy of the first step both in the suggested and advanced setup is a little different
- The notice in the first step of the advanced setup was removed

#### Screenshots and videos

- Before (recommended setup)

<img width="1077" alt="Screen Shot 2022-02-14 at 16 25 52" src="https://user-images.githubusercontent.com/5324818/153932387-dcd64e89-a0da-4f3d-b61d-d5d27f5f0f01.png">

- Before (advanced setup)

<img width="1094" alt="Screen Shot 2022-02-14 at 16 28 37" src="https://user-images.githubusercontent.com/5324818/153932792-54539667-3f0a-48d3-b14c-9a77cee2840d.png">

- After (recommended setup)

https://user-images.githubusercontent.com/5324818/153932448-b59aeada-be21-461d-8221-9d14fec20a93.mp4

- After (advanced setup)

https://user-images.githubusercontent.com/5324818/153933723-c971a00e-bdc9-4918-b1b2-15114cfb4a42.mp4

### Testing instructions

- Build this branch locally or open the live Calypso link
- Select a site which has a connected domain
- Open the connection setup page for that domain by accessing the URL `/domains/mapping/<site_slug>/setup/<domain>`
- Ensure the setup flow look like the videos above and that the cited changes are present
- Ensure the breadcrumbs take you to the correct pages
